src/io/iohk/scalanet/peergroup/DTLSPeerGroup.scala

```diff
@@ -0,0 +1,242 @@
+      dtlsConnector.send(rawData)
+      Task.fromFuture(promisedSend.future)
+    }
+
+    override def close(): Task[Unit] = {
+      val id = (dtlsConnector.getAddress, to.inetSocketAddress)
+      activeChannels(id).in.onComplete()
+      activeChannels.remove(id)
+      Task.unit
+    }
+  }
+
+  private class ClientChannelImpl(to: InetMultiAddress, dtlsConnector: DTLSConnector)(implicit codec: Codec[M])
+      extends ChannelImpl(to, dtlsConnector) {
+    override def close(): Task[Unit] = {
+      dtlsConnector.stop()
+      dtlsConnector.destroy()
+      super.close()
+    }
+  }
+
+  private def createClientConnector(): DTLSConnector = {
+    val connectorConfig = config.scandiumConfigBuilder
+      .setAddress(new InetSocketAddress(config.processAddress.inetSocketAddress.getAddress, 0))
+      .setClientOnly()
+      .build()
+
+    val connector = new DTLSConnector(connectorConfig)
+
+    connector.setRawDataReceiver(new RawDataChannel {
+      override def receiveData(rawData: RawData): Unit = {
+        val channelId = (connector.getAddress, rawData.getInetSocketAddress)
+
+        assert(activeChannels.contains(channelId), s"Missing channel for channelId $channelId")
+
+        val activeChannel: ChannelImpl = activeChannels(channelId)
+
+        val messageE = codec.decode(ByteBuffer.wrap(rawData.bytes))
+
+        messageE.foreach(message => activeChannel.in.onNext(message))
+      }
+    })
+
+    connector
+  }
+
+  private def createServerConnector(): DTLSConnector = {
+    val connectorConfig = config.scandiumConfigBuilder.build()
+
+    val connector = new DTLSConnector(connectorConfig)
+
+    connector.setRawDataReceiver(new RawDataChannel {
+      override def receiveData(rawData: RawData): Unit = {
+        val channelId = (processAddress.inetSocketAddress, rawData.getInetSocketAddress)
+
+        val activeChannel: ChannelImpl = activeChannels.getOrElseUpdate(channelId, createNewChannel(rawData))
+
+        val messageE = codec.decode(ByteBuffer.wrap(rawData.bytes))
+
+        messageE.foreach(message => activeChannel.in.onNext(message))
+      }
+
+      private def createNewChannel(rawData: RawData): ChannelImpl = {
+        val newChannel = new ChannelImpl(InetMultiAddress(rawData.getInetSocketAddress), connector)
+        channelSubject.onNext(newChannel)
+        newChannel
+      }
+    })
+
+    connector
+  }
+}
+
+object DTLSPeerGroup {
+
+  val supportedCipherSuites = Seq(
+    TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+    TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8,
+    TLS_ECDHE_ECDSA_WITH_AES_256_CCM_8,
+    TLS_ECDHE_ECDSA_WITH_AES_128_CCM,
+    TLS_ECDHE_ECDSA_WITH_AES_256_CCM,
+    TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,
+    TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256,
+    TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384
+  )
+
+  trait Config {
+    val bindAddress: InetSocketAddress
+    val processAddress: InetMultiAddress
+    private[scalanet] def scandiumConfigBuilder: DtlsConnectorConfig.Builder
```

Comment:
there are a lot of setting in this `DtlsConnectorConfig.Builder` , it would be nice to have at least documentation of all of them, or maybe case class which captures all of them with some defaula. As a user of library it would great to be able configure most of connection properties, without need to change library code.

Reply:
Yeah, of course there are loads of configuration options that a production implementation might need. I plan to drive the addition of all of these from real use-cases.

What I am keen to avoid is spending a lot of time developing a full-featured, error free, DTLS implementation and then find we don't use it, or we only ever use it in one configuration, etc.

Reply:
A simple way is to let the user provide the `DtlsConnectorConfig` or the builder.